scripts/migrate_from_givtcp.py

```
@@ -616,13 +780,42 @@ async def run(args: argparse.Namespace) -> int:
+        print("     un-migrated. Re-run without --apply to inspect the full table.")
+        await ws.close()
+        return 2
+
     if applying:
         print()
         print("  ⚠️  This will CLEAR and REWRITE long-term statistics for the")
         print("  ⚠️  listed givenergy_local entities. Back up your recorder")
         print("  ⚠️  database before proceeding.")
-        print("  ℹ️  Re-running (same or different --cutover) is safe: the script")
-        print("  ℹ️  always reads the original GivTCP entity for pre-cutover data.")
+        print("  ℹ️  Re-running with the SAME or a LATER --cutover is safe and")
```

Ruff (0.15.15)

[warning] String contains ambiguous `ℹ` (INFORMATION SOURCE). Did you mean `i` (LATIN SMALL LETTER I)?

(RUF001)